### PR TITLE
Device event listing

### PR DESCRIPTION
**Pull request by hardiesoft**
- Adds a device event listing for super admins on browse-next
  - Adds end-points to get known event-types per device in a time window, so 
- Tidies up the locations page
- Temp fix to add location to audio recordings for devices with known locations
- Pulls in UI changes for browse for device sync interface
- Display out of date software packages on device diagnostics page
  - Adds an end-point to get latest software versions of camera software across models and release channels.
- Small fixes for issues with device battery listing
- Fix latest active times for Stations/locations

### CODE REVIEW

**Comment by hardiesoft** [comment]
@Zainrax Just pulled in your latest changes and merged them with mine.

File: api/api/V1/Device.ts [hunk 1]
```diff
@@ -460,6 +462,23 @@ export default function (app: Application, baseUrl: string) {
...
+    `${apiUrl}/latest-software-versions`,
+    extractJwtAuthorizedUser,
+    async function (request: Request, response: Response) {
+      const result = await (
+        await fetch(
+          "https://raw.githubusercontent.com/TheCacophonyProject/salt-version-info/main/salt-version-info.json"
+        )
+      ).json();
+      logger.info("!!!!");
+      console.log(result);
```
Comment (Zainrax) [inline api/api/V1/Device.ts]: minor cleanup

File: browse-next/src/views/DeviceDiagnosticsSubView.vue [hunk 2]
```diff
@@ -327,7 +323,7 @@ const initBatteryInfoTimeSeries = () => {
     }));
     if (batteryAll.length) {
```
Comment (Zainrax) [inline browse-next/src/views/DeviceDiagnosticsSubView.vue:326]: Should filter out null y values from batteryAll

File: browse/src/api/Device.api.ts [hunk 3]
```diff
@@ -263,27 +263,39 @@ async function getType(
...
+      location: LatLng;
+    }>
+  >;
 }
 
 async function updateDeviceSettings(
   deviceId: DeviceId,
   settings: ApiDeviceHistorySettings
 ): Promise<FetchResult<{ settings: ApiDeviceHistorySettings }>> {
+  debugger;
```
Comment (Zainrax) [inline browse/src/api/Device.api.ts]: I will clean this

File: api/api/fileUploaders/uploadGenericRecording.ts
Comment (Zainrax) [inline api/api/fileUploaders/uploadGenericRecording.ts:1]: @hardiesoft you would know better than me, but would it not be easier to use the file signature/magic number from the byte array to determine the file type? Unlikely it needs a change, just wanted to understand the logic/reasoning

**Review by Zainrax — APPROVED** [review]
Looks great :+1: A few minor comments

**Pull request merged**